projects/plugins/jetpack/modules/theme-tools/site-logo/inc/functions.php

```diff
@@ -155,20 +174,7 @@ function jetpack_the_site_logo() {
 	 */
 	echo apply_filters(
 		'jetpack_the_site_logo',
-		sprintf(
-			'<a href="%1$s" class="site-logo-link" rel="home" itemprop="url">%2$s</a>',
-			esc_url( home_url( '/' ) ),
-			wp_get_attachment_image(
-				$logo_id,
-				$size,
-				false,
-				array(
-					'class'     => "site-logo attachment-$size",
-					'data-size' => $size,
-					'itemprop'  => 'logo',
-				)
-			)
-		),
+		$logo_html,
```

Comment:
I pause with how drastic the filtered content becomes. With an ID, there's a full HTML block. Without it, we're echoing out an empty string.

What about earlier in the function, checking `jetpack_has_site_logo` and returning early if there is none present?

Lastly, do you know how Twenty Sixteen is bringing in the Jetpack logo? We may be better off to switch that to use Core's since WP.com and anything that modern Jetpack could be installed on would have Core's site logo? This may be a bad idea too depending on if Core's classes are different (thus possibly breaking custom CSS?)

Reply:
> What about earlier in the function, checking jetpack_has_site_logo and returning early if there is none present?

The presence of the `jetpack_the_site_logo` filter suggests to me, that the original intention here was to provide a way to modify the final logo output, irrespective of an actual logo being set.

An example scenario here would be, adding a placeholder logo for example.

Returning early, if logo is not set would break this.

> Lastly, do you know how Twenty Sixteen is bringing in the Jetpack logo? We may be better off to switch that to use Core's since WP.com and anything that modern Jetpack could be installed on would have Core's site logo? This may be a bad idea too depending on if Core's classes are different (thus possibly breaking custom CSS?)

I'm reluctant to be making such changes at this point, as I am not fully aware of the various setups over a number of versions and legacy code that might depend on loading one type of logo over the other.

Reply:
In that case, what about a conditional so that if there is a logo id, the image is included in the markup. If there isn't, the markup is passed with an empty string where the image would be.

Would that satisfy the bug?

Reply:
I don't understand the reasoning behind returning 'empty/broken' markup when logo id is not set. Could you please elaborate?

Reply:
We'd be changing the output of the filter from what it's done since it was first shipped. I don't know how people may be using the filter so if we can satisfy the bug without changing the filter output, that's what I'm thinking.

Reply:
Right. Thanks for explaining.

There is an issue with the generation of the markup in case logo id doesn't exist.

`wp_get_attachment_image()` doesn't produce an expected output and I'd like to avoid hardcoding HTML. Ideas?

Reply:
Apologies for the delay with the holidays.

That's fair. I suppose if we announce this in the changelog, we'd be okay changing the filter output.

I reviewed all of the instances of plugins and themes with more than 400 installs in the w.org repo using `jetpack_the_site_logo`. I didn't see any that added to that hook.

On WordPress.com, there are a number of premium themes that use the filter, but they all look to wrap what's provided by the filter in an additional div—not checking or doing anything with the value given to them by the filter. 

I didn't check to see what this would do for those themes nor the usage numbers.

What do you think about that?